cirq/contrib/acquaintance/executor.py

```diff
@@ -17,21 +17,25 @@
 import abc
 from collections import defaultdict
 
-from cirq import circuits, ops
-from cirq.contrib.acquaintance.gates import AcquaintanceOpportunityGate
+from cirq import circuits, devices, ops
+
+from cirq.contrib.acquaintance.gates import (
+        AcquaintanceOpportunityGate, ACQUAINT)
 from cirq.contrib.acquaintance.devices import (
         is_acquaintance_strategy)
 from cirq.contrib.acquaintance.permutation import (
         PermutationGate,
-        LogicalIndex,
+        LogicalIndex, LogicalIndexSequence,
         LogicalGates, LogicalMapping)
-from cirq.contrib.acquaintance.strategy import (
+from cirq.contrib.acquaintance.mutation_utils import (
         expose_acquaintance_gates)
 
+
 if TYPE_CHECKING:
     # pylint: disable=unused-import
     from typing import Callable, List, DefaultDict
 
+
 class ExecutionStrategy(metaclass=abc.ABCMeta):
```

Comment:
This class looks like it might be abstraction for abstraction's sake. Is there a reason not to downgrade it into a `NamedTuple` or just a small set of operations accepted by various methods?

Reply:
It's nice to be able to have execution-strategy-specific methods; see `GreedyExecutionStrategy`.